homeassistant/components/zha/core/cluster_handlers/__init__.py

```diff
@@ -501,6 +510,26 @@ async def _get_attributes(
 
     get_attributes = functools.partialmethod(_get_attributes, False)
 
+    async def write_attributes_safe(
+        self, attributes: dict[str, Any], manufacturer: int | None = None
+    ) -> None:
+        """Wrap `write_attributes` to throw an exception on attribute write failure."""
+
+        res = await self.write_attributes(attributes, manufacturer=manufacturer)
+
+        for record in res[0]:
+            if record.status != Status.SUCCESS:
+                try:
+                    name = self.cluster.attributes[record.attrid].name
+                    value = attributes.get(name, "unknown")
```

Comment:
Sorry, it seems my comment never got posted!

This can raise a `KeyError` in two circumstances:

1. If a device decides to respond with the wrong ID (e.g. you read 0x1234, the device says "0x5678 doesn't exist").
2. If we have no definition for an attribute.